model/map-hot-rod/src/main/java/org/keycloak/models/map/storage/hotRod/IckleQueryOperators.java

```diff
@@ -82,17 +84,29 @@
         String combine(String fieldName, Object[] values, Map<String, Object> parameters);
     }
 
-    private static String exists(String modelField, Object[] values, Map<String, Object> parameters) {
-        String field = C + "." + modelField;
+    private static String exists(String modelFieldName, Object[] values, Map<String, Object> parameters) {
+        String field = C + "." + modelFieldName;
         return field + " IS NOT NULL AND " + field + " IS NOT EMPTY";
     }
 
-    private static String notExists(String modelField, Object[] values, Map<String, Object> parameters) {
-        String field = C + "." + modelField;
+    private static String notExists(String modelFieldName, Object[] values, Map<String, Object> parameters) {
+        String field = C + "." + modelFieldName;
         return field + " IS NULL OR " + field + " IS EMPTY";
     }
 
-    private static String in(String modelField, Object[] values, Map<String, Object> parameters) {
+    private static String iLike(String modelFieldName, Object[] values, Map<String, Object> parameters) {
+        String sanitizedValue = (String) IckleQueryMapModelCriteriaBuilder.sanitize(values[0]);
+        return singleValueOperator(ModelCriteriaBuilder.Operator.ILIKE)
```

Comment:
This smells with a possibility to introduce an infinite loop in the future.
```suggestion
        return singleValueOperator(ModelCriteriaBuilder.Operator.LIKE)
```

Reply:
Why do you think so? In `singleValueOperator` the ILIKE operator is exchanged for "LIKE" string using `IckleQueryOperators.operatorToString(op)`.

Reply:
You are correct. Currently, this works like this. What this method does is that it delegates the creation of the resulting query to another operator (LIKE) with a different parameter - lowercase. However, it is now using `ILIKE`, which works because `ILIKE` is mapped to the same operator string. 

I just find it a little bit confusing that `ILIKE` is used again. This is just a nitpick so feel free to ignore it.